add enhancement with child census vectors and future functionality for tax data

mountainMath opened the pull request:
fixes https://github.com/mountainMath/cancensus/issues/134 and adds support for future tax data release

dshkol commented on R/census_vectors.R:
```diff
@@ -0,0 +1,221 @@
+#' Query the CensusMapper API for available vectors for a given dataset.
```
Which part adds support for the tax releases?

mountainMath replied:
New aggregation type and units in list_census_vectors to help with new metadata and variable types.

dshkol commented on R/census_vectors.R:
```diff
@@ -0,0 +1,221 @@
+#' list_census_vectors("CA16") %>%
+#'   filter(vector == "v_CA16_4092") %>%
+#'   parent_census_vectors()
+parent_census_vectors <- function(vector_list){
```
This should try to work the same way as child census vectors. Right now:

```
list_census_vectors("CA16") %>%
  filter(vector == "v_CA16_4092") %>%
  parent_census_vectors()
```
Works as usual, but
```
  parent_census_vectors("v_CA16_4092")
```
Throws a very odd error
```
Error in handle_cm_status_code(response, NULL) : 
  Download of Census Data failed. <!DOCTYPE html>
<html>
<head>
  <title>The page you were looking for doesn't exist (404)</title>
  <meta name="viewport" content="width=device-width,initial-scale=1">
  <style>
  body {
    background-color: #EFEFEF;
    color: #2E2F30;
    text-align: center;
    font-family: arial, sans-serif;
    margin: 0;
  }

  div.dialog {
    width: 95%;
    max-width: 33em;
    margin: 4em auto 0;
  }

  div.dialog > div {
    border: 1px solid #CCC;
    border-right-color: #999;
    border-left-color: #999;
    border-bottom-color: #BBB;
    border-top: #B00100 solid 4px;
    border-top-left-radius: 9px;
    border-top-right-radius: 9px;
    background-color: white;
    padding: 7px 12% 0;
    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
  }

  h1 {
    font-size: 100%;
    color: #730E15;
    line-height: 1.5em;
  }

  div.dialog > p {
    margin: 0 0 1em;
    padding: 1em;
    background-color: #F7F7F7;
    border: 1px solid #CCC;
    border-right-
```
If we don't want the same functionality, should at least add better error handling here.

mountainMath replied:
good catch, did not update the parent_census_vectors call. will do that.

dshkol commented on R/census_vectors.R:
```diff
@@ -0,0 +1,221 @@
+#' list_census_vectors("CA16") %>%
+#'   filter(vector == "v_CA16_4092") %>%
+#'   child_census_vectors(TRUE)
+child_census_vectors <- function(vector_list, leaves_only=FALSE,max_level=NA){
```
Tested both old and new method here, and works nicely. The examples need to be updated though with the new functionality, also that particular vector doesn't have children oops

mountainMath commented:
Ok, fixed the parent_census_vectors call. And moved the census_regions stuff into a separate file to make things cleaner. Also changed the caching behaviour of list_census_regions and list_census_vectors to cache for duration of running R session by default. Should make things run smoother.

dshkol commented:
Added some tweaks to the documentation and examples that were bothering me or were in need of an update for new syntax.

dshkol commented:
If there's no CRAN check issues or notes (I had one, but it seemed specific to my local directory), happy to push this one into master and move to submit the update to CRAN

mountainMath commented:
CRAN check looks good on my end. Only other thing I can think of is adding tests. Maybe leave that for the next release? 😁

dshkol commented:
Should be added, but I think can wait for the next one.

dshkol commented:
Let's merge and submit then?
